service/storage_service.cc

```diff
@@ -2960,11 +2960,11 @@ future<> storage_service::restore_replica_count(inet_address endpoint, inet_addr
     if (is_repair_based_node_ops_enabled(streaming::stream_reason::removenode)) {
         auto ops_uuid = node_ops_id::create_random_id();
         auto ops = seastar::make_shared<node_ops_info>(ops_uuid, nullptr, std::list<gms::inet_address>());
-        return _repair.local().removenode_with_repair(get_token_metadata_ptr(), endpoint, ops).finally([this, notify_endpoint] () {
+        co_return co_await _repair.local().removenode_with_repair(get_token_metadata_ptr(), endpoint, ops).finally([this, notify_endpoint] () {
```

Comment:
Please don't mix coroutines with continuations. You should unwrap the `finally()` clause here too.

Reply:
why not?
unwrapping the finally clause meaning catching the exception, or getting removenode_with_repair as_future,
and handling later, while .finally is simpler.

Reply:
It's not good practice and can lead to weird bugs.

Reply:
Fixed in v2